ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/handler/TaskHandler.java

```diff
@@ -121,7 +122,8 @@ protected ValidReaderWriteIdList getValidCleanerWriteIdList(CompactionInfo info,
     // been some delta/base dirs
     assert rsp != null && rsp.getTblValidWriteIdsSize() == 1;
 
-    return TxnCommonUtils.createValidReaderWriteIdList(rsp.getTblValidWriteIds().get(0));
+    return new ValidCleanerWriteIdList(
+        TxnCommonUtils.createValidReaderWriteIdList(rsp.getTblValidWriteIds().get(0)));
```

Comment:
What about creating a separate method like createValidCleanerWriteIdList(..)?

Reply:
it's just used in 1 single place, i would prefer not to create 1 more util method